Refactor the Plugin JS

joemcgill opened the pull request:
This cleans up the plugin JS by abstracting the logic for updating the `srcset` and `sizes` attributes into their own functions. In order to grab all the available sizes for `srcset` attributes (see #80), I've added an our own AJAX function which will use the plugin internals to retrieve `srcset` values and pass them back to the editor instead of duplicating all of that logic in the JS.

tevko commented:
I'm with it

jaspermdegroot commented:
I tested and reviewed and it looks good to me.

One minor thing though. The styling of the comments (starting with capital, period at the end) and the indentation (tabs instead of two spaces) don't match with the WordPress JS coding standards.

joemcgill commented:
Added a few bits of cleanup, including the .editorconfig file from WP core in order to help with coding style enforcement. Added and fixed a few docs as well.
